controllers/imagerepository_controller.go

```diff
@@ -113,6 +116,19 @@ func (r *ImageRepositoryReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 }
 
 func (r *ImageRepositoryReconciler) scan(ctx context.Context, imageRepo imagev1alpha1.ImageRepository, ref name.Reference) (imagev1alpha1.ImageRepository, error) {
+	// acquire lock
```

Comment:
I don't understand why this is necessary. The controller runtime ensured that Reconcile doesn't get run with the same resource concurrently, doesn't it?

Reply:
It doesn't, without the lock you can receive multiple updates events and they will run in parallel.

Reply:
To test this, increase the workers to 2 or more and run multiple kubectl apply, the repo will be scanned in parallel.

Reply:
Not processing the same resource concurrently is a documented feature of the workqueue: https://godoc.org/k8s.io/client-go/util/workqueue

Reply:
Hmm doesn't this apply to single worker? We decided to use the lock after getting parallel calls in helm operator.

Reply:
The workqueue should prevent concurrency across workers according to https://openkruise.io/en-us/blog/blog2.html

So as long as we don't branch to other go routines should be safe to remove the lock.